src/Canvas.cc

```diff
@@ -758,6 +758,11 @@ Canvas::ResolveFontDescription(const PangoFontDescription *desc) {
     for (; it != _font_face_list.end(); ++it) {
       if (g_ascii_strcasecmp(families[i], pango_font_description_get_family(it->user_desc)) == 0) {
         char *name = g_strdup(pango_font_description_get_family(it->sys_desc));
+
+        if (!name) {
```

Comment:
Thanks again for another bug fix.

I think it can be caught more early on during registration, when the `sys_desc` is built. In [`get_pango_font_description`](https://github.com/Automattic/node-canvas/blob/master/src/register_font.cc#L199) I think it should be bailing early [here](https://github.com/Automattic/node-canvas/blob/master/src/register_font.cc#L209) (plus memory cleanup), otherwise there's a null font description present which would be nice to avoid altogether.

That'd also throw the `"Could not parse font file"` error which people could handle.

Reply:
I've attached another commit, is it better now?